backend/modules/metadata_store/prisma_store.py

```diff
@@ -132,60 +131,78 @@ async def alist_collections(self) -> List[str]:
 
     async def adelete_collection(self, collection_name: str, include_runs=False):
         try:
-            deleted_collection: Optional[
-                "PrismaCollection"
-            ] = await self.db.collection.delete(where={"name": collection_name})
-            if not deleted_collection:
-                raise HTTPException(
-                    status_code=404,
-                    detail=f"Failed to delete collection {collection_name!r}. No such record found",
-                )
-            if include_runs:
-                try:
-                    _deleted_count = await self.db.ingestionruns.delete_many(
-                        where={"collection_name": collection_name}
+            # Initialize a prisma transaction
+            async with self.db.tx() as transaction:
+                # Delete all collections by name
+                deleted_collection: Optional[
+                    "PrismaCollection"
+                ] = await transaction.collection.delete(where={"name": collection_name})
+                # If the collection is not found, raise an error
+                if not deleted_collection:
```

Comment:
I was not able to find a proper error in docs if delete_collection is performed on nonexistent record. Will check more into this in Prisma python client source code.